ember-core/src/main/scala/org/http4s/ember/core/Parser.scala

```diff
@@ -209,18 +209,26 @@ private[ember] object Parser {
               EmberException.ParseError("Invalid Empty Init Line")),
           Applicative[F].pure(_)
         )
+        _ <- logger.trace(s"HeadersSection - ${headersBV.decodeAscii}")
         headers <- generateHeaders(headersBV)(Headers.empty)(logger)
+        _ <- logger.trace(show"Headers: $headers")
+
         (httpV, status) <- bvToResponseTopLine[F](methodHttpUri)
 
-        contentLength = headers.get(org.http4s.headers.`Content-Length`).map(_.length).getOrElse(0L)
+        _ <- logger.trace(s"HttpVersion: $httpV - Status: $status")
+
+        contentLength = headers.get(org.http4s.headers.`Content-Length`).map(_.length)
+        transferEncoding = headers.get(org.http4s.headers.`Transfer-Encoding`)
```

Comment:
Nit: you shouldn't have both of these and can probably defer looking for it and save a few nanoseconds per request.